cmd/ignited/cmd/gitops.go

```diff
@@ -85,15 +106,25 @@ func addGitOpsFlags(fs *pflag.FlagSet, f *gitOpsFlags) {
 	fs.DurationVar(&f.timeout, "timeout", f.timeout, "Git operation (clone, push, pull) timeout")
 
 	fs.StringVar(&f.identityFile, "identity-file", f.identityFile, "What SSH identity file to use for pushing")
+	fs.StringVar(&f.hostsFile, "hosts-file", f.hostsFile, "What known_hosts file to use for remote verification")
 	fs.StringVar(&f.username, "https-username", f.username, "What username to use when authenticating with Git over HTTPS")
 	fs.StringVar(&f.password, "https-password", f.password, "What password/access token to use when authenticating with Git over HTTPS")
 
 	// TODO: We need to add path prefix support to the WatchStorage to support this
 	// fs.StringSliceVarP(&f.paths, "paths", "p", f.paths, "What subdirectories to care about. Default the whole repository")
+}
 
-	// TODO: When https://github.com/fluxcd/toolkit/issues/2 is fixed and the
-	// https://github.com/fluxcd/source-controller/tree/master/internal/crypto/ssh/knownhosts package
-	// can be vendored, we'll add support for reading a hosts file. In the meantime, the
-	// SSH_KNOWN_HOSTS variable can be used.
-	// fs.StringVar(&f.hostsFile, "hosts-file", f.hostsFile, "What hosts file to use")
+// resolveHomeEnv resolves the ~ character in file paths at runtime
+func resolveHomeEnv(path string) (string, error) {
+	// exit quickly if possible
+	if !strings.Contains(path, "~") {
+		return path, nil
+	}
+	// get the HOME env var and fail if not found
+	homeEnv := os.Getenv("HOME")
+	if len(homeEnv) == 0 {
+		return "", fmt.Errorf("Couldn't lookup home directory for user")
+	}
+	// replace the first tilde in the path with $HOME
+	return strings.Replace(path, "~", homeEnv, 1), nil
```

Comment:
This will perform an illicit replace if the ~ is not the first character of the string. We should likely do something like this (which also retrieves the home directory correctly): https://stackoverflow.com/a/43578461 if this function is really needed (see comment above)

Reply:
true. using https://github.com/mitchellh/go-homedir/blob/master/homedir.go now :+1: